kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/dhcp/DhcpClientLeases.java

```diff
@@ -0,0 +1,170 @@
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.kura.KuraErrorCode;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.net.IPAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class defines DHCP leases
+ *
+ */
+public class DhcpClientLeases {
+
+    private static final Logger logger = LoggerFactory.getLogger(DhcpClientLeases.class);
+
+    private static final String GLOBAL_DHCP_LEASES_DIR = "/var/lib/dhcp";
+    private static final String IFACE_DHCP_LEASES_DIR = "/var/lib/dhclient";
+
+    private static DhcpClientLeases dhcpClientLeases = null;
+
+    public static synchronized DhcpClientLeases getInstance() {
+        if (dhcpClientLeases == null) {
+            dhcpClientLeases = new DhcpClientLeases();
+        }
+        return dhcpClientLeases;
+    }
+
+    public List<IPAddress> getDhcpGateways(String interfaceName, String address) throws KuraException {
+        return getDhcpGateways(interfaceName, parseHostAddress(address));
+    }
+
+    public List<IPAddress> getDhcpGateways(String interfaceName, IPAddress address) throws KuraException {
+        List<IPAddress> gateways = new ArrayList<>();
+        if (interfaceName == null || interfaceName.isEmpty() || address == null || address.getAddress() == null) {
+            return gateways;
+        }
+
+        List<DhcpClientLeaseBlock> leaseBlocks = getLeaseBlocks(getDhclientFile(interfaceName));
+        for (DhcpClientLeaseBlock leaseBlock : leaseBlocks) {
+            if (!leaseBlock.matches(interfaceName, address)) {
+                continue;
+            }
+            List<IPAddress> routers = leaseBlock.getRouters();
+            for (IPAddress router : routers) {
+                if (!gateways.contains(router)) {
+                    gateways.add(router);
+                }
+            }
+        }
+        return gateways;
+    }
+
+    public List<IPAddress> getDhcpDnsServers(String interfaceName, String address) throws KuraException {
+        return getDhcpDnsServers(interfaceName, parseHostAddress(address));
+    }
+
+    public List<IPAddress> getDhcpDnsServers(String interfaceName, IPAddress address) throws KuraException {
+
+        List<IPAddress> servers = new ArrayList<>();
+        if (interfaceName == null || interfaceName.isEmpty() || address == null || address.getAddress() == null) {
+            return servers;
+        }
+
+        List<DhcpClientLeaseBlock> leaseBlocks = getLeaseBlocks(getDhclientFile(interfaceName));
+        for (DhcpClientLeaseBlock leaseBlock : leaseBlocks) {
+            if (!leaseBlock.matches(interfaceName, address)) {
+                continue;
+            }
+            List<IPAddress> dnsServers = leaseBlock.getDnsServers();
+            for (IPAddress dnsServer : dnsServers) {
+                if (!servers.contains(dnsServer)) {
+                    servers.add(dnsServer);
+                }
+            }
```

Comment:
same as above

Reply:
Yes, I did this intentionally even though lease blocks appear to have the same information. I thought that if it is not, I will collect all DNS servers.